Kernel/FileSystem/FIFO.h

```diff
@@ -32,11 +32,12 @@ class FIFO final : public File {
     ErrorOr<NonnullRefPtr<OpenFileDescription>> open_direction(Direction);
     ErrorOr<NonnullRefPtr<OpenFileDescription>> open_direction_blocking(Direction);
 
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Woverloaded-virtual"
+    // Ignore warnings about overloading virtual File::{at,de}tach(OpenFileDescription&)
+    using File::attach;
+    using File::detach;
+
     void attach(Direction);
     void detach(Direction);
```

Comment:
🤔 overloaded-virtual is a nice warning imo. we should probably rename the overloads that take a direction so that we can't possibly confuse them with the virtual methods we are not intending to override in this class.

Reply:
`attach` had a single call site, so I decided to inline it. I could remove `detach(Direction)` by overriding `detach(OpenFileDescription&)` for `FIFO` to also decrement the reader/writer count.